### DISCUSSION TITLE
Site Auto Refresh

### DISCUSSION THREAD

**Discussion by nkostrencic** [sqlpage/SQLPage#54] [Q&A] [answered]
Greetings,

First and foremost - great product, thanks for creating this!

I'm creating a data processing pipeline with a wizard using the steps component. I'm wondering if there is a good way to create an auto refresh for a page?

Basically refresh the status shown on a page until processing is done and send the user to a results page. I had an idea of using pg_sleep as part of a redirect (redirect to the same page) component query to postpone a redirect for a certan amount of time, redirect to the results page when processing done, however, the redirect component can be used only before it returns content so I cannot really show anything to the user.

Another idea would be to build a custom component with <meta http-equiv="refresh" content="10"> or something along those lines. Any other ideas?

P.S. I'm building a system you can use to view data in a database (tables, graphs etc.) and creating a few pages that can be used to kick off a data refresh by calling AWS Lambdas from an Aurora PG database through a query.

**Comment by lovasoa** [comment, marked as answer]
Yes, creating an auto-refreshing page is easy, you can use the `refresh` property of the [`shell` component](https://sql.ophir.dev/documentation.sql?component=shell#component).

```sql
select 'shell' as component, 3 as refresh
```

Have a look at this example from one of the demos for an advanced example: https://github.com/lovasoa/SQLpage/blob/main/examples/corporate-conundrum/wait.sql

**Reply by nkostrencic** [reply]
Thanks for the quick reply. Will dig deeper into that example!
